routing/router.go

```diff
@@ -1966,28 +1968,38 @@ func (r *ChannelRouter) sendPayment(payment *LightningPayment,
 // getFailedChannelID tries to locate the failing channel given a route and the
 // pubkey of the node that sent the error. It will assume that the error is
 // associated with the outgoing channel of the error node.
-func getFailedChannelID(route *Route, errSource *btcec.PublicKey) (
+func getFailedChannelID(route *Route, errSource Vertex) (
 	uint64, error) {
 
-	// As this error indicates that the target channel was unable to carry
-	// this HTLC (for w/e reason), we'll query the index to find the
-	// _outgoing_ channel the source of the error was meant to pass the
-	// HTLC along to.
-	if badChan, ok := route.nextHopChannel(errSource); ok {
-		return badChan.ChannelID, nil
+	// If the error originates from ourselves, report our outgoing channel
+	// as failing.
+	if errSource == route.SourcePubKey {
+		return route.Hops[0].ChannelID, nil
 	}
 
-	// If we weren't able to find the hop *after* this node, then we'll
-	// attempt to disable the previous channel.
-	//
-	// TODO(joostjager): errSource must be the final hop then? In that case,
-	// certain types of errors are not expected. For example
-	// FailUnknownNextPeer. This could be a reason to prune the node?
-	if prevChan, ok := route.prevHopChannel(errSource); ok {
-		return prevChan.ChannelID, nil
+	hopCount := len(route.Hops)
+	for i, hop := range route.Hops {
+		if errSource == hop.PubKeyBytes {
```

Comment:
style nit: `continue` in case `errSource != hop.PubKeyBytes` to avoid indentatin.

Reply:
fixed